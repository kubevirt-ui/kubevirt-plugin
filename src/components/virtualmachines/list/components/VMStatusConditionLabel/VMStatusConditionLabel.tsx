import * as React from 'react';

import { V1VirtualMachineCondition } from '@kubevirt-ui/kubevirt-api/kubevirt';
import { Label, LabelGroup, Popover, PopoverPosition } from '@patternfly/react-core';

export const VMStatusConditionLabel: React.FC<V1VirtualMachineCondition> = React.memo(
  (condition) => {
    return (
      <Popover
        position={PopoverPosition.top}
        aria-label="Condition Popover"
        bodyContent={() => <div>{condition?.message}</div>}
      >
        <Label
          color="grey"
          isTruncated
          href="#"
          onClick={(e) => {
            e.preventDefault();
          }}
        >
          {condition?.reason}={condition?.status}
        </Label>
      </Popover>
    );
  },
);
VMStatusConditionLabel.displayName = 'VMStatusConditionLabel';

export const VMStatusConditionLabelList: React.FC<{ conditions: V1VirtualMachineCondition[] }> =
  React.memo(({ conditions }) => {
    return (
      <LabelGroup>
        {conditions.map(({ message, reason, status, type }) => (
          <VMStatusConditionLabel
            key={type}
            message={message}
            reason={reason}
            status={status}
            type={type}
          />
        ))}
      </LabelGroup>
    );
  });
VMStatusConditionLabelList.displayName = 'VMStatusConditionLabelList';
