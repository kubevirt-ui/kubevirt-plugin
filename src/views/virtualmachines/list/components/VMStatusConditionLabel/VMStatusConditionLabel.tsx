import * as React from 'react';

import { V1VirtualMachineCondition } from '@kubevirt-ui/kubevirt-api/kubevirt';
import { Label, LabelGroup, Popover, PopoverPosition } from '@patternfly/react-core';

export const VMStatusConditionLabel: React.FC<V1VirtualMachineCondition> = React.memo(
  (condition) => {
    return (
      <Popover
        aria-label="Condition Popover"
        bodyContent={() => <div>{condition?.message ?? condition?.reason}</div>}
        position={PopoverPosition.top}
      >
        <Label
          onClick={(e) => {
            e.preventDefault();
          }}
          color="grey"
          href="#"
          isTruncated
        >
          {condition?.type}={condition?.status}
        </Label>
      </Popover>
    );
  },
);
VMStatusConditionLabel.displayName = 'VMStatusConditionLabel';

export const VMStatusConditionLabelList: React.FC<{ conditions: V1VirtualMachineCondition[] }> =
  React.memo(({ conditions }) => {
    return (
      <LabelGroup>
        {conditions?.map(({ message, reason, status, type }) => (
          <VMStatusConditionLabel
            key={type}
            message={message}
            reason={reason}
            status={status}
            type={type}
          />
        ))}
      </LabelGroup>
    );
  });
VMStatusConditionLabelList.displayName = 'VMStatusConditionLabelList';
