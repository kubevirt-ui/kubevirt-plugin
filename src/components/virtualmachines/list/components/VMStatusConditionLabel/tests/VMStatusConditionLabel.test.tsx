import * as React from 'react';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import { VMStatusConditionLabel } from '../VMStatusConditionLabel';

import { conditionsMock } from './mocks';

afterEach(cleanup);

test('VMStatusConditionLabel', async () => {
  const { message, reason, status, type } = conditionsMock[0];
  const { asFragment, getByText } = render(
    <VMStatusConditionLabel message={message} reason={reason} status={status} type={type} />,
  );
  const firstRender = asFragment();

  expect(firstRender).toMatchSnapshot();

  // click on condition to open popover
  fireEvent.click(getByText(`${reason}=${status}`));
  const popoverMessage = await screen.findByText(message);

  expect(popoverMessage).toHaveTextContent(message);
});
