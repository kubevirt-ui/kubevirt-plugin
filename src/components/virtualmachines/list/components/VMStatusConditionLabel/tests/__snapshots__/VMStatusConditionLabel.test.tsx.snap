// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VMStatusConditionLabel 1`] = `
<DocumentFragment>
  <span
    class="pf-c-label"
  >
    <a
      class="pf-c-label__content"
      href="#"
    >
      <span
        class="pf-c-label__text"
      >
        no_vmi=true
      </span>
    </a>
  </span>
</DocumentFragment>
`;
